src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Helpers/UriBuilderExtensions.cs

```diff
@@ -47,5 +50,32 @@ public static void AppendQueryParameters(this UriBuilder builder, string queryPa
                 builder.Query = queryParams;
             }
         }
+
+        public static string GetHttpsUriWithOptionalPort(string host, string tenant, string path, int port)
+        {
+            var builder = new UriBuilder("https", host);
+            builder.Path = string.Format(CultureInfo.InvariantCulture, "{0}/{1}", tenant, path);
+
+            //No need to set port if it equals 443 as it is the default https port
+            if (port != DefaultHttpsPort)
+            {
+                builder.Port = port;
+            }
```

Comment:
I wonder why you pass in host to the UriBuilder as you are overriding the path... thoughts?

Reply:
when ADAL uses the authority, it strips out the host and appends various paths to it to create the various endpoints. I was simply following that logic. Other than that there is no real concrete reason.